File: spatial_field_model/examine_spatial_data.m

%
% Look at the station data with respect to current topography.
%

% Read geographic position of grid points
lon = ncread('../real-data/wrfout_d03_latlon.nc', 'XLONG');
lon = lon(:, :, 1);
lat = ncread('../real-data/wrfout_d03_latlon.nc', 'XLAT');
lat = lat(:, :, 1);

% Read in the variables
T = ncread('../real-data/wrfout_d03_T2.nc', 'T2');
Q = ncread('../real-data/wrfout_d03_Q2.nc', 'Q2');
P = ncread('../real-data/wrfout_d03_PSFC.nc', 'PSFC');

% compute moisture equilibrium field
[Ed, Ew] = equilibrium_moisture(P, Q, T);

% visualize this for all time steps
for i=1:size(Ed,3)
    subplot(121);
    imagesc(Ed(:, end:-1:1, i)');
    title(sprintf('Drying, frame %d', i));
    caxis([0, 0.55]);
    colorbar();
    subplot(122);
    imagesc(Ew(:, end:-1:1, i)');
    title(sprintf('Wetting, frame %d', i));
    caxis([0, 0.55]);
    colorbar();
    pause(0.2);
end

File: spatial_field_model/examine_spatial_data.py
# -*- coding: utf-8 -*-
"""
Created on Wed Oct 24 17:40:24 2012

@author: martin
"""

from spatial_model_utilities import load_wrf_data, equilibrium_moisture, render_spatial_field

import numpy as np
from mpl_toolkits.basemap import Basemap
import matplotlib.pyplot as plt



if __name__ == '__main__':
    
    # we will refer to these often
    v = load_wrf_data('../real_data/california/realfire05_d03_20120409.nc')
#    v = load_wrf_data('../real_data/witch_creek/realfire03_d02_20071021.nc')

    # read in vars
    lat = v['XLAT'][0,:,:]
    lon = v['XLONG'][0,:,:]
    rain = v['RAINNC']
    Q2 = v['Q2']
    T2 = v['T2']
    P = v['PSFC']

    # construct our basemap
    lat_rng = (np.min(lat), np.max(lat))
    lon_rng = (np.min(lon), np.max(lon))
    m = Basemap(llcrnrlon=lon_rng[0],llcrnrlat=lat_rng[0],
                urcrnrlon=lon_rng[1],urcrnrlat=lat_rng[1],
                projection = 'mill')

    # compute the equilibrium moisture on grid points
    Ed, Ew = equilibrium_moisture(v['PSFC'][:,:,:], Q2, v['T2'][:,:,:])

    plt.ion()
    plt.figure(figsize = (10, 6))
    for i in range(Ed.shape[0]):
        if i > 0 and np.all(rain[i,:,:] == 0):
            continue

        plt.subplot(221)
        render_spatial_field(m, lon, lat, Ed[i,:,:], 'Drying equilibrium [%d]' % i)
        plt.clim([np.min(Ed), np.max(Ed)])
        if i == 0:
            plt.colorbar()
        plt.subplot(222)
        render_spatial_field(m, lon, lat, Ew[i,:,:], 'Wetting equilibrium')
        plt.clim([np.min(Ew), np.max(Ew)])
        if i == 0:
            plt.colorbar()
        plt.subplot(223)
        render_spatial_field(m, lon, lat, rain[i,:,:], 'Rain')
        plt.clim([np.min(rain), np.max(rain)])
        if i == 0:
            plt.colorbar()
        plt.subplot(224)
        render_spatial_field(m, lon, lat, Q2[i,:,:], 'Water vapor ratio')
        plt.clim([np.min(Q2), np.max(Q2)])
        if i == 0:
            plt.colorbar()
        plt.draw()

    plt.ioff()
